create-pull-request.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * Small Utility script to help us creating a Pull Request using Github's Rest API.
+ */
+
+$user = getenv('GITHUB_ACTOR');
+$token = getenv('GITHUB_TOKEN');
+$repository = getenv('GITHUB_REPOSITORY');
+$body = file_get_contents('php://stdin');
+
+['h' => $head] = getopt('h:');
+
+if (empty($user)) {
+    throw new RuntimeException('GITHUB_ACTOR not set!');
+}
+
+if (empty($token)) {
+    throw new RuntimeException('GITHUB_TOKEN not set!');
+}
+
+if (empty($repository)) {
+    throw new RuntimeException('GITHUB_REPOSITORY not set!');
+}
+
+if (empty($body)) {
+    throw new RuntimeException('Update message empty!');
+}
+
+if (empty($head)) {
+    throw new RuntimeException('Head branch not specified!');
+}
+
+$ch = curl_init();
+curl_setopt($ch, CURLOPT_URL, "https://api.github.com/repos/$repository/pulls");
+curl_setopt($ch, CURLOPT_POST, 1);
+curl_setopt($ch, CURLOPT_USERAGENT, 'Drupal7-Security-Updates');
+curl_setopt($ch, CURLOPT_USERPWD, "$user:$token");
+curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode(
+  [
+    'title' => 'Automated Security Updates',
+    'body' => $body,
+    'head' => $head,
+    'base' => 'master',
+    'labels' => ['security update']
+  ]
+));
+curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+$output = curl_exec($ch);
+if ($output === false) {
+    echo 'Curl error';
+    echo curl_error($ch);
+    curl_close($ch);
+
+    exit(1);
+}
+curl_close($ch);
+$output = json_decode($output);
+
+if (empty($output)) {
+    throw new RuntimeException('Empty Response!');
+}
+
+if(!empty($output->html_url)) {
+    echo $output->html_url;
+    exit();
+}
+
+/*
+ * If there are any errors other than that the Pull Request already exists, we
+ * throw an error.
+ *
+ * It's not easy to check if a PR for a branch already exists, so, instead, we
+ * simply always try to create one and ignore the error.
+ */
+if(!empty($output->errors)) {
+    $repoOwner = explode('/', $repository)[0];
+    foreach ($output->errors as $error) {
+        if (empty($error->message)) {
+            throw new RuntimeException('Unknown API error!');
+        }
+        if($error->message !== "A pull request already exists for $repoOwner:$head.") {
+            throw new RuntimeException('Error creating pull request: ' . $error->message);
+        }
+    }
+}
```

Comment:
Why create a whole script for this when there's an action that can be used?
https://github.com/marketplace/actions/create-pull-request

Or, is it that the action can't be used?

Reply:
Mainly for 2 reasons:
- Security: Internally, any action that creates a Pull Request will need a token that gives write access to the repo. I don't like the idea of giving access to that code to some code written by someone else, even though it will be a temporary token created for a single run of the workflow
- Simplicity: We'll need add security update workflows to a dozen different repos, to the fewer steps we have in the workflow, the less we'll need to duplicate

Reply:
Ummmm, you know your OS was written by someone else, right? 😆 

It's ok to include this script. The only question I have (as a side note) is: how are we going to import the action into our own repos? Copying it or using the marketplace?

Reply:
> how are we going to import the action into our own repos? Copying it or using the marketplace?

Neither. It's just going to be used an regular action. See [here](https://github.com/compucorp/outdated-d7-site/blob/master/.github/workflows/security-updates.yml#L28) for an example (Using `@master` there because we don't have a tag yet)